cinnamon-session/main.c

```diff
@@ -383,8 +386,26 @@ main (int argc, char **argv)
          */
         csm_util_setenv ("GNOME_DESKTOP_SESSION_ID", "this-is-deprecated");
 
-        /* Make QT5 apps follow the GTK style */
-        csm_util_setenv ("QT_STYLE_OVERRIDE", "gtk");
+
+        /* Make QT5 apps follow the GTK style. Starting with QT 5.7, a different
+         * env var has to be set than what worked in previous versions.
+         */
+        qt_platform_theme_current = g_getenv ("QT_QPA_PLATFORMTHEME");
+        qt_style_override_current = g_getenv ("QT_STYLE_OVERRIDE");
+        qt_platform_theme_new = HAVE_QT57 ? "qt5ct" : "qgnomeplatform";
+
+        if (NULL == qt_platform_theme_current) {
+            csm_util_setenv ("QT_QPA_PLATFORMTHEME", qt_platform_theme_new);
+        }
+
+        if (NULL == qt_style_override_current) {
+            csm_util_setenv ("QT_STYLE_OVERRIDE", "gtk");
+        }
+
+        g_free(qt_platform_theme_current);
+        g_free(qt_style_override_current);
+        g_free(qt_platform_theme_new);
```

Comment:
I guess it segfaults here - there's no need to free `qt_platform_theme_new` as the memory wasn't allocated dynamically.